### PR TITLE
Disallow global bots from bypassing blockpms

### PR DESCRIPTION
**Pull request by SSlayer9**
https://www.smogon.com/forums/threads/disallow-global-bots-from-bypassing-blocked-pms-or.3666899/
this ^. If one of the alternative solutions is better lmk.

### CODE REVIEW

**Comment by SSlayer9** [comment]
check failed which is odd, i tested this and it worked

File: server/chat.ts [hunk 1]
```diff
@@ -989,7 +989,7 @@ export class CommandContext extends MessageContext {
 				}
 				if (targetUser.settings.blockPMs &&
 					(targetUser.settings.blockPMs === true || !user.authAtLeast(targetUser.settings.blockPMs)) &&
-					!user.can('lock')) {
+					(!user.can('lock' || user.group === '*'))) {
```
Comment (AnnikaCodes) [inline server/chat.ts:992]:
```suggestion
					(!user.can('lock') || user.group === '*') {
```

Comment (AnnikaCodes) [reply server/chat.ts]: Personally, I'd implement this by adding a `bypassBlockPMs` permission rather than hardcoding `*` rank.

Comment (AnnikaCodes) [reply server/chat.ts]: (Also, you should test your changes locally before opening a pull request -- that way, you could have caught the error I fixed in my suggestion.)

File: server/chat.ts [hunk 2]
```diff
@@ -989,7 +989,7 @@ export class CommandContext extends MessageContext {
 				}
 				if (targetUser.settings.blockPMs &&
 					(targetUser.settings.blockPMs === true || !user.authAtLeast(targetUser.settings.blockPMs)) &&
-					!user.can('lock')) {
+					(!user.can('lock') || user.group === '*') {
```
Comment (AnnikaCodes) [inline server/chat.ts:992]:
```suggestion
					(!user.can('lock') || user.group === '*')) {
```
that's my fault, I'm bad at reading :)

Comment (Zarel) [reply server/chat.ts:992]: Global bots can't lock, though...

Comment (Zarel) [reply server/chat.ts:992]: Is this actually the check? If so, in theory, nothing else needs to be done, bot PMs are already blockable.

Comment (AnnikaCodes) [reply server/chat.ts:992]: I can confirm that global bots don't bypass blocked PMs on main. This PR can probably be closed along with the suggestion thread.

**Comment by Zarel** [comment]
Thanks for the attempt, at least.

**Pull request closed without merging**